tsdb/functions.go

```diff
@@ -60,15 +61,21 @@ func initializeMapFunc(c *influxql.Call) (mapFunc, error) {
 	case "distinct":
 		return MapDistinct, nil
 	case "sum":
-		return MapSum, nil
+		return func(itr Iterator) interface{} {
```

Comment:
Would it be possible to abstract the parsing of arguments from the `Call` object. 

For example `MapSum` sums the values of a field. It needs the call object to know which field to sum.

Could we make the `MapSum` signature 
```func MapSum(itr Iterator, field string) interface{}```
And then extract the field from the `Call` object here in this anon func. And just pass an empty field if it doesn't exist.

```
return func(itr Iterator) interface{} {
        var  field string
        if fields := c.Fields(); len(fields) > 0 {
                field = fields[0]
        }
        return MapSum(itr, field)
}, nil
```

A quick glance at the uses of `Call` in the functions shows that they are used in a similar manner and that the logic is not dependent on the data in the itr.